packages/js-client/src/context.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  SimpleStorageContextParams,
+  SimpleStorageContextState,
+  SimpleStorageOverridenState,
+} from "./types";
+import { Context, ContextCore } from "@aragon/sdk-client-common";
+
+// set your defaults here or import them from a package
+const DEFAULT_SIMPLE_STORAGE_PLUGIN_ADDRESS =
+  "0x1234567890123456789012345678901234567890";
+const DEFAULT_SIMPLE_STORAGE_Repo_ADDRESS =
+  "0x2345678901234567890123456789012345678901";
+
+export class SimpleStorageContext extends ContextCore {
+  // super is called before the properties are initialized
+  // so we initialize them to the value of the parent class
+  protected state: SimpleStorageContextState = this.state;
+  protected overriden: SimpleStorageOverridenState = this.overriden;
+  constructor(
+    contextParams?: Partial<SimpleStorageContextParams>,
+    aragonContext?: Context,
+  ) {
+    // parent contructor will call this.set(contextParams)
+    // so we don't need to call it again
+    super(contextParams);
+    if (aragonContext) {
+      // copy the context properties to this
+      Object.assign(this, aragonContext);
+    }
+    if (contextParams) {
+      // overide the context params with the ones passed to the constructor
+      this.set(contextParams ?? {});
```

Comment:
Not needed since we are within the "if" block

```suggestion
      this.set(contextParams);
```